nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestListSFTP.java

```diff
@@ -74,12 +82,14 @@ public void setUp() throws Exception {
     @After
     public void tearDown() throws Exception {
         sshServer.stopServer();
+        Files.deleteIfExists(Paths.get(sshServer.getVirtualFileSystemPath()));
     }
 
     @Test
-    public void testRunFileFound() {
+    public void testRunFileFound() throws InterruptedException {
+        writeTempFile(1);
+        Thread.sleep(STABILITY_WAIT);
```

Comment:
It would be helpful to track down potential issues as opposed to introducing `Thread.sleep()`, which creates more test performance problems.

Reply:
I've been trying to figure out what I can do about this. The issue is that I will intermittently see '        runner.assertTransferCount(ListSFTP.REL_SUCCESS, 1);' fail with a result of 0.

I've tried:
- Asserting that the file count available in the virtual SSH/SFTP server matches the number created for the test (the files are present to be listed)
- Asserting that the runner is valid
- Asserting that the controller services are valid

I've currently got it working, rather than using a thread sleep, by setting the runner's scheduler to 500ms instead, and running 2 iterations... If that's not acceptable, I may need to perform the test in TestAbstractListProcessor. However it seems like it could be useful to increase the testing coverage for TestListSFTP.

Reply:
Thanks for the update. Running multiple iterations might work, although that seems questionable.

Since the listing strategy handling is not part of `ListSFTP` itself, it seems like these tests would be better run in the context of the class that implements the logic. Not sure if that just shifts the problem to a different location, but it might be worth considering.

Reply:
The `ListSFTP` processor includes a file filter that checks the minimum age based on the current system time in milliseconds. The `SFTPTransfer.newFileInfo()` method sets the last modified time, but apparently it only provides resolution to the second and includes a multiplication by 1000 to get milliseconds. With the default minimum age set to zero seconds, this might explain the reason that the test sometimes misses files.

With that background, another solution may be to set the minimum age to a negative number.

Reply:
Yes I think you're correct, I think the file age was the key problem. I'll submit an update to the PR in a few moments.